serviceconnector/src/deleteLinker/deleteLinker.ts

```diff
@@ -3,20 +3,21 @@
 *  Licensed under the MIT License. See License.md in the project root for license information.
 *--------------------------------------------------------------------------------------------*/
 
-import { AzExtTreeItem, AzureWizard, AzureWizardExecuteStep, AzureWizardPromptStep, DeleteConfirmationStep, ExecuteActivityContext, IActionContext, createSubscriptionContext, isAzExtTreeItem } from "@microsoft/vscode-azext-utils";
+import { AzureWizard, AzureWizardExecuteStep, AzureWizardPromptStep, DeleteConfirmationStep, ExecuteActivityContext, IActionContext, ISubscriptionContext, createSubscriptionContext } from "@microsoft/vscode-azext-utils";
+import { AzureSubscription } from "@microsoft/vscode-azureresources-api";
 import * as vscode from 'vscode';
-import { LinkerItem } from "../createLinker/createLinker";
+import { isAzureSubscription } from "../createLinker/createLinker";
 import { DeleteLinkerStep } from "./DeleteLinkerStep";
 import { IPickLinkerContext } from "./IPickLinkerContext";
 import { LinkerListStep } from "./LinkerListStep";
 
-export async function deleteLinker(context: IActionContext & ExecuteActivityContext, item: LinkerItem | AzExtTreeItem, preSteps: AzureWizardPromptStep<IPickLinkerContext>[] = []): Promise<void> {
-    const subscription = isAzExtTreeItem(item) ? item.subscription : createSubscriptionContext(item.subscription); // For v1.5 compatibility
+export async function deleteLinker(context: IActionContext & ExecuteActivityContext, id: string, subscription: ISubscriptionContext | AzureSubscription, serviceConnectorName?: string, preSteps: AzureWizardPromptStep<IPickLinkerContext>[] = []): Promise<void> {
```

Comment:
Is there any reason for serviceConnectorName to be optional? It's a bit weird, but it seems like this would have a no-op if for some reason it was undefined. If that happens, we should probably throw an error otherwise the user wouldn't know that nothing happened.

Reply:
It is optional because when the command palette is used you have to choose a specific service connection (LinkerListStep) and in this case serviceConnectorName wouldn't be passed.

Reply:
I see.  Hmm, though you could get rid of the if conditional. If serviceConnectorName is undefined, you'd just be setting linkerName to undefined, which it probably already was that.